### PR TITLE
Allow modifying kernel associated with a session via PATCH

### PR DESCRIPTION
**Pull request by minrk**
- PATCH to kernel.id attaches to other kernel
- PATCH to kernel.name starts new kernel with given name

Previous kernel is shutdown after the change

cc @blink1073

### CODE REVIEW

File: notebook/services/sessions/tests/test_sessions_api.py [hunk 1]
```diff
@@ -115,10 +123,47 @@ def test_delete(self):
...
+    def test_modify_path(self):
         newsession = self.sess_api.create('foo/nb1.ipynb').json()
         sid = newsession['id']
 
-        changed = self.sess_api.modify(sid, 'nb2.ipynb').json()
+        changed = self.sess_api.modify_path(sid, 'nb2.ipynb').json()
         self.assertEqual(changed['id'], sid)
         self.assertEqual(changed['notebook']['path'], 'nb2.ipynb')
+
+    def test_amodify_kernel_name(self):
```
Comment (blink1073) [inline notebook/services/sessions/tests/test_sessions_api.py]: (sp.)

File: notebook/services/sessions/tests/test_sessions_api.py [hunk 2]
```diff
@@ -115,10 +123,47 @@ def test_delete(self):
...
+        self.assertEqual(after['notebook'], before['notebook'])
+        self.assertNotEqual(after['kernel']['id'], before['kernel']['id'])
+
+        # check kernel list, to be sure previous kernel was cleaned up
+        r = requests.get(url_path_join(self.base_url(), 'api/kernels'))
+        r.raise_for_status()
+        kernel_list = r.json()
+        self.assertEqual(kernel_list, [after['kernel']])
+
+    def test_amodify_kernel_id(self):
```
Comment (blink1073) [inline notebook/services/sessions/tests/test_sessions_api.py]: (sp.)

**Comment by blink1073** [comment]
:+1:, thanks Min!

File: notebook/services/sessions/tests/test_sessions_api.py (hunk 2 above)
Comment (minrk) [inline notebook/services/sessions/tests/test_sessions_api.py]: Fixed, thanks

**Comment by blink1073** [comment]
I confirmed this works for me on upcoming PRs to jupyter-js-services and jupyter-js-notebook.

File: notebook/services/sessions/handlers.py [hunk 3]
```diff
@@ -86,19 +86,46 @@ def get(self, session_id):
...
             notebook = model['notebook']
             if 'path' in notebook:
                 changes['path'] = notebook['path']
+        if 'kernel' in model:
+            if 'name' in model['kernel']:
+                kernel_name = model['kernel']['name']
+                kernel_id = yield sm.start_kernel_for_session(
+                    session_id, kernel_name=kernel_name, path=before['notebook']['path'])
+                changes['kernel_id'] = kernel_id
+            if 'id' in model['kernel']:
```
Comment (blink1073) [inline notebook/services/sessions/handlers.py]: I'm sending `null` if the `id` is not given, do you mind checking for that here?

Comment (minrk) [inline notebook/services/sessions/handlers.py]: Good call. Done.

**Comment by blink1073** [comment]
@minrk, see also the spelling error above.

**Pull request merged**